fix: trezord kill on app exit

mroz22 opened the pull request:
Found that there were 2 places stopping `before-quit` events from executing. 

-  when calling `mainWindow.getBounds()` inside `before-quite` event. Doing this feels wrong in general as this event is fired after `window-all-closed` so doing something with browserWindow instance which maybe (I am not sure yet, did not check that yet) does not exist anymore doesn't feel right. 
-  to my dismay, also calling `electronLocalshortcut.unregisterAll(mainWindow);` in `before-quit` stops further `before-quit` events from executing. I don't know why yet, but my first question is whether it is needed to unregister some event listeners for key presses when app is going to stop after few miliseconds anyway? 

As it is now, it should be fixing #4904 and also it fixes a bug (not sure if there is an issue) that window size is not saved and therefore can't be restored on next application run.

mroz22 commented on packages/suite-desktop/src-electron/app.ts:
```diff
@@ -83,6 +83,17 @@ const init = async () => {
+            logger.debug('app', 'new winBounds saved');
+        }, 1000);
+    });
+
```
not sure if debounce is necessary. `resize` event does not fire so often. but maybe it is "better" to do some debouncing here.

matejkriz replied:
I believe debounce is a good idea here, but you have to check if `mainWindow` is still defined before trying to getBounds() and would be nice to clear that timeout on `closed`

mroz22 replied:
added [ca43b7a](https://github.com/trezor/trezor-suite/pull/4931/commits/ca43b7ad5e56f29b157ed50c7e0b3da7f72082e9)

matejkriz commented on packages/suite-desktop/src-electron/modules/shortcuts.ts:
```diff
@@ -23,9 +22,11 @@ const init = ({ mainWindow, src }: Dependencies) => {
-    app.on('before-quit', () => {
-        electronLocalshortcut.unregisterAll(mainWindow);
-    });
+    // todo: calling electronLocalshortcut.unregisterAll(mainWindow) has unpleasant side-effect
```
I believe that all local shortcuts should be unregistered at that point already by `electron-localshortcut` itself

mroz22 replied:
removed commented code in [12baf63](https://github.com/trezor/trezor-suite/pull/4931/commits/12baf635bfab1e99ce895baf5c4b9f3d968e627d)

matejkriz reviewed (approved):
works for me (arm mac) 👌

mroz22 commented:
squashed, rebased and waiting for QA to aproved this before merging. 
So far I believe these have been tested and work:
- [x] linux  
- [x] mac
- [x] windows

wendys-cats commented:
QA OK - tested on Win10 and macOS (intel).

Could not replicate the issue.
